src/harness/unittests/tsserverProjectSystem.ts

```diff
@@ -6178,6 +6182,13 @@ namespace ts.projectSystem {
                 timeoutAfterReloadFs = true;
                 verifyAfterPartialOrCompleteNpmInstall(2);
 
+                if (!timeoutDuringPartialInstallation) {
+                    const project = projectService.configuredProjects.get(tsconfigJson.path);
+                    for (const err of project.getCurrentProgram().getSemanticDiagnostics()) {
+                        assert.fail(`Unexpected semantic error ${err.messageText} in project`);
+                    }
+                }
+

```

Comment:
Instead of changing this test case can we add separate test case that verifies actual scenario of not invalidating module resolution correctly for subFolder/app2 (by verifying semanticDiagnostics) for both classic and node modules resolution